server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -309,6 +311,9 @@ public void validate(Integer numRoutingShards, Map<Setting<Integer>, Integer> se
 
     private final Index index;
     private final long version;
+
+    private final long mappingVersion;
```

Comment:
I wonder whether `MappingMetaData` would be a better place to maintain the mapping version. The reason it feels like a better place to me is that this version is about an instance of `MappingMetaData`.

Reply:
I considered this but it ends up that we construct mapping metadata in quite a few places where having the mapping version is not natural. We do not hold on to the mapping metadata in the mapping service, so in some of these places it means we would have to hold on to the mapping version in the mapper service. But then we would have multiple places in the system carrying the mapping version. Because of this I considered the index metadata to be better. Additionally, like that we have the builder for the index metadata. Let me know if you feel strongly about this, or see a good way to have it on the mapping metadata.

Reply:
I don't feel strongly about this. I was just wondering about this. Your explanation is a good reason why to add the version to IndexMetadata instead of MappingMetaData.